ddtrace/llmobs/_writer.py

```diff
@@ -157,3 +164,112 @@ def enqueue(self, event: LLMObsEvaluationMetricEvent) -> None:
 
     def _data(self, events: List[LLMObsEvaluationMetricEvent]) -> Dict[str, Any]:
         return {"data": {"type": "evaluation_metric", "attributes": {"metrics": events}}}
+
+
+class LLMObsSpanEncoder(BufferedEncoder):
+    content_type = "application/json"
+
+    def __init__(self, *args):
+        super(LLMObsSpanEncoder, self).__init__()
+        self._lock = forksafe.RLock()
+        self._buffer_limit = 1000
+        self._init_buffer()
+
+    def __len__(self):
+        with self._lock:
+            return len(self._buffer)
+
+    def _init_buffer(self):
+        with self._lock:
+            self._buffer = []
+
+    def put(self, events: List[LLMObsSpanEvent]):
+        # events always has only 1 event - with List type to be compatible with HTTPWriter interfaces
+        with self._lock:
+            if len(self._buffer) >= self._buffer_limit:
+                logger.warning(
+                    "%r event buffer full (limit is %d), dropping event", self.__class__.__name__, self._buffer_limit
+                )
+                return
+            self._buffer.extend(events)
+
+    def encode(self):
+        with self._lock:
+            if not self._buffer:
+                return
+            events = self._buffer
+            self._init_buffer()
+        data = {"_dd.stage": "raw", "event_type": "span", "spans": events}
+        try:
+            enc_llm_events = json.dumps(data)
+            logger.debug("encode %d LLMObs span events to be sent", len(events))
+        except TypeError:
+            logger.error("failed to encode %d LLMObs span events", len(events), exc_info=True)
+            return
+        return enc_llm_events
+
+
+class LLMObsEventClient(WriterClientBase):
+    def __init__(self):
+        encoder = LLMObsSpanEncoder(0, 0)
+        super(LLMObsEventClient, self).__init__(encoder)
+
+
+class LLMObsEventProxiedEventClient(LLMObsEventClient):
+    ENDPOINT = EVP_PROXY_AGENT_ENDPOINT
+
+
+class LLMObsSpanAgentWriter(HTTPWriter):
```

Comment:
@Yun-Kim when we implement retries for the agentless writer we'll likely want to refactor this so that we only have a single writer class and 
`_configure_span_writer` toggles the endpoint + headers used, thoughts?

Reply:
Will address in separate PRs.

Reply:
Yes, exactly. cc @tomshen